aws-xray-recorder-sdk-aws-sdk/src/main/java/com/amazonaws/xray/handlers/TracingHandler.java

```diff
@@ -194,15 +193,8 @@ public void beforeRequest(Request<?> request) {
         currentSubsegment.setNamespace(Namespace.AWS.toString());
 
         if (recorder.getCurrentSegment() != null && recorder.getCurrentSubsegment().shouldPropagate()) {
-            boolean isSampled = previousSubsegment.isPresent() ?
-                    previousSubsegment.get().isSampled() :
-                    recorder.getCurrentSegment().isSampled();
-
-            TraceHeader header =
-                new TraceHeader(recorder.getCurrentSegment().getTraceId(),
-                        isSampled ? currentSubsegment.getId() : null,
-                        isSampled ? SampleDecision.SAMPLED : SampleDecision.NOT_SAMPLED);
-            request.addHeader(TraceHeader.HEADER_KEY, header.toString());
+            request.addHeader(TraceHeader.HEADER_KEY, TraceHeader.fromEntity(
```

Comment:
This code doesn't look equivalent to the one it's replacing.
In the left side code, the parent id as the TraceHeader argument is either `currentSubsegment`'s id or `null`. Only the `isSampled` is fetched from either previousSubsegment or the `recorder.getCurrentSegment()`.
In the right side code, the whole TraceHeader will be created either from `previousSubsegment` or `currentSubsegment`.

Reply:
Wow, good catch! This was a bug.
I looked at the original code and I don't think we even need to use the previous subsegment here.